Выполнение ТЗ-11

krist-losev opened the pull request.

krist-losev committed:
Выполнение ТЗ-11

krist-losev committed:
Выполнение ТЗ-11

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,63 +1,60 @@
+        return filmService.updateFilm(newFilm);
+    }
+
+    //найти фильм по айди
```
Вы уже выучили достаточно много, поэтому предлагаю уйти от однострочных комментариев в сторону документирующих:
```
/**
* Поиск фильма по id.
* @id идентификатор фильма.
*/
```

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,63 +1,60 @@
+
+    //найти фильм по айди
+    @GetMapping("/{id}")
+    public Optional<Film> findFilmById(@PathVariable int id) {
```
Используя Optional в качестве ответа, вы переносите логику на клиента, что чаще всего некорректно. Кроме того, вы успешно возвращаете как наличие объекта, так и его отсутствие. У вас есть ErrorHandler, который обрабатывает случай отсутствия объекта. Поэтому следует воспользоваться им

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,62 +1,66 @@
+
+    //поиск пользователя по id
+    @GetMapping("/{postId}")
+    public Optional<User> findUserById(@PathVariable int postId) {
```
Здесь также следует поправить

kriwsha commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,67 @@
+
+    private FilmStorage filmStorage;
+    private UserStorage userStorage;
+    private static final Comparator<Film> LIKES_FILM = Comparator.comparing(film -> film.getLike().size(), Comparator.reverseOrder());
```
Статические переменные следует выносить вверх класса

krist-losev committed:
Исправление замечаний ревьюера:
1) исправление в поиске пользователя/фильма по идентификтору
2) исправление комментариев
